pkg/apis/config/sigstore_keys.go

```diff
@@ -133,56 +87,171 @@ func NewSigstoreKeysFromConfigMap(config *corev1.ConfigMap) (*SigstoreKeysMap, e
+		return nil // TODO: log error? Add return error?
+	}
+	logID, err := cosign.GetTransparencyLogID(pk)
+	if err != nil {
+		return nil // TODO: log error? Add return error?
 	}
+
+	return &pbtrustroot.TransparencyLogInstance{
+		BaseUrl:       source.BaseURL.String(),
+		HashAlgorithm: HashStringToHashAlgorithm(source.HashAlgorithm),
+		PublicKey:     pbpk,
+		LogId: &pbcommon.LogId{
+			KeyId: []byte(logID),
+		},
+	}
+}
+
+func HashStringToHashAlgorithm(hash string) pbcommon.HashAlgorithm {
+	switch hash {
+	case "sha-256", "sha256":
+		return pbcommon.HashAlgorithm_SHA2_256
+	case "sha-384", "sha384":
+		return pbcommon.HashAlgorithm_SHA2_384
+	case "sha-512", "sha512":
+		return pbcommon.HashAlgorithm_SHA2_512
+	default:
+		return pbcommon.HashAlgorithm_HASH_ALGORITHM_UNSPECIFIED
+	}
+}
+
+func SerializeCertChain(certChain *pbcommon.X509CertificateChain) []byte {
+	var chain []byte
+	for _, cert := range certChain.Certificates {
+		bytes := cert.RawBytes
+		block := &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: bytes,
+		}
+		chain = append(chain, pem.EncodeToMemory(block)...)
+	}
+	return chain
+}
+
+func SerializePublicKey(publicKey *pbcommon.PublicKey) []byte {
+	block := &pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: publicKey.RawBytes,
+	}
+	return pem.EncodeToMemory(block)
+}
+
+func DeserializeCertChain(chain []byte) *pbcommon.X509CertificateChain {
+	var certs []*pbcommon.X509Certificate
+	for {
+		var block *pem.Block
+		block, chain = pem.Decode(chain)
+		if block == nil {
+			break
+		}
+		certs = append(certs, &pbcommon.X509Certificate{RawBytes: block.Bytes})
+	}
+	return &pbcommon.X509CertificateChain{Certificates: certs}
+}
+
+func DeserializePublicKey(publicKey []byte) (*pbcommon.PublicKey, crypto.PublicKey, error) {
+	block, _ := pem.Decode(publicKey)
+	if block == nil {
+		return nil, nil, fmt.Errorf("failed to decode public key")
+	}
+	pk, err := cryptoutils.UnmarshalPEMToPublicKey(publicKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal public key: %w", err)
+	}
+	var keyDetails pbcommon.PublicKeyDetails
+	switch k := pk.(type) {
+	case *ecdsa.PublicKey:
+		switch k.Curve {
+		case elliptic.P256():
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_ECDSA_P256_SHA_256
+		case elliptic.P384():
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_ECDSA_P384_SHA_384
+		case elliptic.P521():
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_ECDSA_P521_SHA_512
+		default:
+			keyDetails = pbcommon.PublicKeyDetails_PUBLIC_KEY_DETAILS_UNSPECIFIED
+		}
+	case *rsa.PublicKey:
+		switch k.Size() {
+		case 2048:
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_RSA_PSS_2048_SHA256
+		case 3072:
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_RSA_PSS_3072_SHA256
+		case 4096:
+			keyDetails = pbcommon.PublicKeyDetails_PKIX_RSA_PSS_4096_SHA256
+		default:
+			keyDetails = pbcommon.PublicKeyDetails_PUBLIC_KEY_DETAILS_UNSPECIFIED
+		}
+	default:
+		keyDetails = pbcommon.PublicKeyDetails_PUBLIC_KEY_DETAILS_UNSPECIFIED
+	}
```

Comment:
I need to spend a few moments to confirm these assumptions with regard to key type. @kommendorkapten, if you have a moment to weigh in, that would be helpful! This code path is exercised when a transparency log public key is provided in a custom resource (PEM encoded), or when a TUF repo does not have a `trusted_root.json`. So it does not apply to the Sigstore Public Good instance, which provides a `trusted_root.json`. For private transparency logs, we need to assume the key type to fill in the values required by the TrustedRoot.

Reply:
I think this code is solid. It uses Go std `ParsePKIXPublicKey`, and the created public key should so have the correct parameters.

The only thing that possibly could break is for RSA is the intended signature scheme is PKCS1#v1.5, but without any OOB information, we can't tell which signature scheme to use.

Do you know where in the code we load the verifier for RSA key? I can also take a look, but there are two options in sigstore/sigstore:
PKCS#1v1.5: https://github.com/sigstore/sigstore/blob/main/pkg/signature/rsapkcs1v15.go#L121
PSS: https://github.com/sigstore/sigstore/blob/main/pkg/signature/rsapss.go#L147

Reply:
Ok, so I think we should change things here.

There are two places I found where a verifier is loaded 
* https://github.com/sigstore/policy-controller/blob/fc2800a5e9cf28f75c63e5ac6f50e2e468f6b5a2/pkg/webhook/validation.go#L38 
* https://github.com/sigstore/policy-controller/blob/fc2800a5e9cf28f75c63e5ac6f50e2e468f6b5a2/pkg/webhook/validator.go#L839

Both of them calls `sigstore/pkg/signature/verifier::LoadVerifier`. That functions will assume PKCS#1v1.5 padding. To use PSS padding, you need to explicitly load that via `signature.LoadRSAPSSVerifier`, there are no such references in this code base.

So in the code above, you should load:
* `PKIX_RSA_PKCS1V15_2048_SHA256`
* `PKIX_RSA_PKCS1V15_3072_SHA256`
* `PKIX_RSA_PKCS1V15_4096_SHA256`

IMHO this should be clarified in the docs too, that when using RSA keys, PKCS#1v1.5 padding is only supported.

Reply:
Thanks @kommendorkapten! I will file an issue for adding the ability use a protojson trusted root in the TrustRoot CRD which will make this section of code obsolete, and by the way, these key details are not actually used anywhere in Policy Controller at the moment -- They will only be used when we add the ability to use Sigstore Bundles as OCI artifacts. So this is not a regression and the trusted root support should be complete before we add any support for bundles.